sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -204,6 +204,23 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
     }
   }
 
+  testStandardAndLegacyModes("array of struct") {
```

Comment:
Do we have a test for array of struct of struct?

Reply:
I added a new test case for this in https://github.com/apache/spark/pull/26993/commits/0f1af94c6cfc72651c3209bc0d31ae1bef140988